mayastor/src/bdev/nexus/nexus_rpc.rs

```diff
@@ -130,8 +132,13 @@ pub(crate) fn register_rpc_methods() {
             let key: Option<String> =
                 if args.key == "" { None } else { Some(args.key) };
 
+            let share_proto = ShareProtocolNexus::from_i32(args.share);
+            if share_proto.is_none() {
+                return Err(Error::InvalidShareProtocol {sp_value: args.share as i32});
+            }
+
             let nexus = nexus_lookup(&args.uuid)?;
-            nexus.share(key).await.map(|device_path| PublishNexusReply {
+            nexus.share(share_proto.unwrap(), key).await.map(|device_path| PublishNexusReply {
```

Comment:
If we remove the Option we can remove the unwrap()

Reply:
Fixed, https://github.com/openebs/Mayastor/pull/125/commits/74322e937b211cca53728d59bf35938af0e6c4ed

Reply:
Fixed by https://github.com/openebs/Mayastor/pull/125/commits/74322e937b211cca53728d59bf35938af0e6c4ed